linker-interface/shared/src/test/scala/org/scalajs/linker/interface/StandardConfigFingerprintTest.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import java.net.URI
+
+import org.junit.Test
+import org.junit.Assert._
+
+import Semantics.RuntimeClassNameMapper._
+
+class StandardConfigFingerprintTest {
+  @Test
+  def noFingerprintCollisionCheckIR(): Unit = {
+    val sc1 = StandardConfig().withCheckIR(false)
+    val sc2 = StandardConfig().withCheckIR(true)
+    assertNotEquals(
+      StandardConfig.fingerprint(sc1),
+      StandardConfig.fingerprint(sc2)
+    )
+  }
+
+  @Test
+  def noFingerprintCollisionRelativizeSourceMapBase(): Unit = {
+    val sc1 = StandardConfig().withRelativizeSourceMapBase(None)
+    val sc2 = StandardConfig().withRelativizeSourceMapBase(Some(new URI("a")))
+    assertNotEquals(
+      StandardConfig.fingerprint(sc1),
+      StandardConfig.fingerprint(sc2)
+    )
+  }
+
+  @Test
+  def noFingerprintCollisionESFeatures(): Unit = {
+    val sc1 = StandardConfig().withESFeatures(_.withUseECMAScript2015(true))
+    val sc2 = StandardConfig().withESFeatures(_.withUseECMAScript2015(false))
+    assertNotEquals(
+      StandardConfig.fingerprint(sc1),
+      StandardConfig.fingerprint(sc2)
+    )
+  }
+
+  @Test
+  def noFingerprintCollisionRuntimeClassNameMapper(): Unit = {
+    val sc1 = StandardConfig().withSemantics(
+      _.withRuntimeClassNameMapper(keepAll() andThen discardAll())
+    )
```

Comment:
Indent 4, closing parens on previous line.

Reply:
Indentation is fixed in eb6e308689224c3bbcc9c059e10fa1f02bee3acb.

I haven't fixed the closing parens yet, because I'm a little unclear on what the coding style guideline is here. Is it that things with varargs are allowed to have a closing parens on a single line, but regular functions are not?

```scala
// This happens often in the Scala.js codebase:
val x = Seq(
    thingNumberOne,
    thingNumberTwo,
    // ...
    thingNumberN
)
```

The code style guide is quite strict on where to break.

> A line can be broken after either a `,` or a `(`, or possibly after a binary operator in a long expression.

This wouldn't really allow the above snippet, or long chained calls (which are usually broken on `)`, before `.`).

Reply:
Yes, it's allowed for "collection-like" constructors like that. Typically things with varargs, indeed, can have the closing paren in a separate line (in that case, the first argument should already be on a separate line than the opening `(`.

Reply:
Okay, I see. Thank you for clearing this up :)